Missing support for postgresql:// URI schema designator

When configuring `MB_DB_CONNECTION_URI` with Postgres, it only works with the `postgres://` URI schema designator. The connection fails with `postgresql://`.  Postgres officially supports both: https://www.postgresql.org/docs/9.6/static/libpq-connect.html#AEN45347

For example:

`export MB_DB_CONNECTION_URI=postgres://localhost:5432/metabase`
`java -jar metabase.jar` works fine.

`export MB_DB_CONNECTION_URI=postgresql://localhost:5432/metabase`
`java -jar metabase.jar` fails with:

```
11-03 22:39:53 INFO metabase.db :: Verifying postgresql Database Connection ...
11-03 22:39:53 ERROR metabase.driver :: Failed to connect to database: java.lang.IllegalArgumentException: No implementation of method: :can-connect? of protocol: #'metabase.driver/IDriver found for class: nil
java.lang.AssertionError: Assert failed: Unable to connect to Metabase postgresql DB.
(binding [*allow-potentailly-unsafe-connections* true] (require (quote metabase.driver)) ((resolve (quote metabase.driver/can-connect-with-details?)) engine details))
	at metabase.db$verify_db_connection.invokeStatic(db.clj:336)
	at metabase.db$verify_db_connection.invoke(db.clj:331)
	at metabase.db$setup_db_BANG_.invokeStatic(db.clj:357)
	at metabase.db$setup_db_BANG_.doInvoke(db.clj:349)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at metabase.core$init_BANG_.invokeStatic(core.clj:112)
	at metabase.core$init_BANG_.invoke(core.clj:92)
	at metabase.core$start_normally.invokeStatic(core.clj:195)
	at metabase.core$start_normally.invoke(core.clj:189)
	at metabase.core$_main.invokeStatic(core.clj:280)
	at metabase.core$_main.doInvoke(core.clj:275)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at metabase.core.main(Unknown Source)
```

Metabase version: 0.20.3
Metabase internal database: Postgres

--- a/src/metabase/db.clj
+++ b/src/metabase/db.clj
@@ -49,7 +49,10 @@
   "Parse a DB connection URI like `postgres://cam@localhost.com:5432/cams_cool_db?ssl=true&sslfactory=org.postgresql.ssl.NonValidatingFactory` and return a broken-out map."
   [uri]
   (when-let [[_ protocol user pass host port db query] (re-matches #"^([^:/@]+)://(?:([^:/@]+)(?::([^:@]+))?@)?([^:@]+)(?::(\d+))?/([^/?]+)(?:\?(.*))?$" uri)]
-    (merge {:type     (keyword protocol)
+    (merge {:type     (case (keyword protocol)
+                        :postgres   :postgres
+                        :postgresql :postgres
+                        :mysql      :mysql)
             :user     user
             :password pass
             :host     host

--- a/test/metabase/db_test.clj
+++ b/test/metabase/db_test.clj
@@ -6,11 +6,18 @@
 (tu/resolve-private-vars metabase.db parse-connection-string)
 
 ;; parse minimal connection string
-(expect {:type :postgres :user nil :password nil :host "localhost" :port nil :dbname "toms_cool_db" }
+(expect
+  {:type :postgres, :user nil, :password nil, :host "localhost", :port nil, :dbname "toms_cool_db" }
   (parse-connection-string "postgres://localhost/toms_cool_db"))
 
+;; parse connection string using alternate `postgreql` URI schema
+(expect
+  {:type :postgres, :user nil, :password nil, :host "localhost", :port nil, :dbname "toms_cool_db" }
+  (parse-connection-string "postgresql://localhost/toms_cool_db"))
+
 ;; parse all fields and query string arguments
-(expect {:type :postgres :user "tom" :password "1234" :host "localhost" :port "5432" :dbname "toms_cool_db" :ssl "true" :sslfactory "org.postgresql.ssl.NonValidatingFactory"}
+(expect
+  {:type :postgres, :user "tom", :password "1234", :host "localhost", :port "5432", :dbname "toms_cool_db", :ssl "true", :sslfactory "org.postgresql.ssl.NonValidatingFactory"}
   (parse-connection-string "postgres://tom:1234@localhost:5432/toms_cool_db?ssl=true&sslfactory=org.postgresql.ssl.NonValidatingFactory"))
 
 

